File: core/src/io/github/jerukan/rendering/Drawable.java
package io.github.jerukan.rendering;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import io.github.jerukan.rendering.camera.OrthoCameraWrapper;

/**
 * Any object in the game that is to be drawn.
 */
public interface Drawable {

    void render(SpriteBatch batch);

    boolean inCameraWindow(OrthoCameraWrapper camera);

    void dispose();
}

File: core/src/io/github/jerukan/rendering/Drawable.kt
package io.github.jerukan.rendering

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import io.github.jerukan.rendering.camera.OrthoCameraWrapper

/**
 * Any object in the game that is to be drawn.
 */
interface Drawable {

    fun render(batch: SpriteBatch)

    fun inCameraWindow(camera: OrthoCameraWrapper): Boolean

    fun dispose()
}